module/zfs/zfs_replay.c

```diff
@@ -706,13 +725,18 @@ zfs_replay_write(void *arg1, void *arg2, boolean_t byteswap)
 		if (zp->z_size < eod)
 			zfsvfs->z_replay_eof = eod;
 	}
-
-	written = zpl_write_common(ZTOI(zp), data, length, &offset,
+#ifdef __FreeBSD__
+	ssize_t resid;
+	error = vn_rdwr(UIO_WRITE, ZTOV(zp), data, length, offset,
+	    UIO_SYSSPACE, IO_SYNC, kcred, NOCRED, &resid, curthread);
+#else
+	int written = zpl_write_common(ZTOI(zp), data, length, &offset,
```

Comment:
I thought the idea with `zfs_file` API was that we could wrap `vn_rdwr`?

Reply:
Yes, when accessing another filesystem.  This is for replay so ideally we want to call `zfs_write()` directly but doing so means creating a uio.  One possibility is we could add a `zfs_write_replay()` wrapper to `linux/zfs/zfs_vnops.c` which is platform specific to do the job.  This is essentially all `zpl_write_common()` is doing on Linux.